confidential-data-hub/docs/use-cases/secure-mount-with-aliyun-oss.md

```diff
@@ -66,39 +66,39 @@ Run CDH
 confidential-data-hub
 ```
 
-Prepare a request JSON `storage.json`
+Prepare a request JSON `storage.json`. This example is for aliyun OSS.
 ```json
 {
-    "driver": "",
-    "driver_options": [
-        "alibaba-cloud-oss={\"akId\":\"XXX\",\"akSecret\":\"XXX\",\"annotations\":\"\",\"bucket\":\"<bucket-id>\",\"encrypted\":\"gocryptfs\",\"encPasswd\":\"<PASSWORD>\",\"kmsKeyId\":\"\",\"otherOpts\":\"-o max_stat_cache_size=0 -o allow_other\",\"path\":\"<bucket-path>\",\"readonly\":\"\",\"targetPath\":\"/mnt/aliyun-oss\",\"url\":\"https://oss-cn-beijing.aliyuncs.com\",\"volumeId\":\"\"}"
-    ],
-    "source": "",
-    "fstype": "",
-    "options": [],
+    "volume_type": "alibaba-cloud-oss",
+    "options": {
+        "akId": "XXX",
+        "akSecret": "XXX",
+        "annotations": "",
+        "bucket": "<bucket-id>",
+        "encrypted": "gocryptfs",
+        "encPasswd": "<PASSWORD>",
+        "kmsKeyId": "",
+        "otherOpts": "-o max_stat_cache_size=0 -o allow_other",
+        "path": "<bucket-path>",
+        "readonly": "",
+        "targetPath": "/mnt/aliyun-oss",
+        "url": "https://oss-cn-beijing.aliyuncs.com",
+        "volumeId": ""
+    },
+    "flags": [],
     "mount_point": "/mnt/target-path"
 }
```

Comment:
I hope that this new request format makes much more sense to you. : ) @bpradipt @fitzthum

Reply:
@Xynnn007 indeed :-) 
A related question, can I use the secure mount functionality today to mount S3 or Azure blob volumes inside the container today ? Or code changes will be needed to support different object storages ?

Reply:
Take aliyun oss for example, we might need
1. CDH to impl the plugin. If 1 is achieved, a confidential VM support is ok. The following points are for confidential containers.
2. kata-runtime & kata-agent deliver `secure-mount` metadata to CDH. This is what https://github.com/kata-containers/kata-containers/pull/7965/files did.
3. The CSI driver of S3/Azure blob/other kinds of storages should support deliver mount info to kata as [aliyun-oss does](https://github.com/kubernetes-sigs/alibaba-cloud-csi-driver/blob/master/pkg/oss/coco.go#L94-L100)

Reply:
Thanks @Xynnn007. This makes sense